pkgs/build-support/trivial-builders.nix

```
@@ -541,4 +541,31 @@ rec {
       phases = "unpackPhase patchPhase installPhase";
       installPhase = "cp -R ./ $out";
     };
+
+  /* Checks the command output contains the specified version
+   *
+   * Examples:
+   *
+   * passthru.tests.version = testVersion { package = hello; };
+   *
+   * passthru.tests.version = testVersion {
+   *   package = seaweedfs;
+   *   command = "weed version";
+   * };
+   *
+   * passthru.tests.version = testVersion {
+   *   package = key;
+   *   command = "KeY --help";
+   *   # Wrong '2.5' version in the code. Drop on next version.
+   *   version = "2.5";
+   * };
+   */
+  testVersion =
+    { package,
+      command ? "${package.pname or package.name} --version",
+      version ? package.version,
+    }: runCommand "test-version" { buildInputs = [ package ]; meta.timeout = 60; } ''
```

```suggestion
    }: runCommand "test-version" { nativeBuildInputs = [ package ]; meta.timeout = 60; } ''
```

`buildInputs` is for components (linking etc) and not for running. With cross compilation and in the future with `strictDeps`, these components will not be added to `PATH`.

@Ericson2314 do you think it's feasible to send this to a remote builder in case of cross compilation? Do we an attribute à la `pkgs.buildPackages` for this purpose? (I've tried `hostP{ackages,*}`, `remoteP*`, `testP*`)

You'd need a remote builder, but that's better than not being able to run this at all.